src/backend/executor/execMain.c

```diff
@@ -3170,7 +3170,7 @@ ExecInsert(TupleTableSlot *slot,
 	partslot = reconstructMatchingTupleSlot(slot, resultRelInfo);
 	if (rel_is_heap)
 	{
-		tuple = ExecCopySlotHeapTuple(partslot);
+		tuple = ExecMaterializeSlotHeapTuple(partslot);
```

Comment:
To avoid oid duplicates, should we could add logic here? So that new oid could be generated by heap_insert:
if (resultRelationDesc->rd_rel->relhasoids)
{
        HeapTupleSetOid(heaptuple, InvalidOid);
}

Reply:
At the moment I'm not seeing any OID duplication problems the way we saw them with `ExecCopySlotHeapTuple`. Do you have a code path in mind that could fail?